### DISCUSSION TITLE
command line password recovery

### DISCUSSION THREAD

**Discussion by kusznir** [Nico640/docker-unms#105] [General]
Hello all:
Somehow I managed to forget the password of my UNMS instance.  And the web gui-based password recovery e-mail doesn't work (no e-mail ever arrives, but that's been my experience with the UBNT-released version too).

I found this URL for command line recovery instructions:
https://help.ui.com/hc/en-us/articles/115015690087-UISP-Command-Line-Interface-CLI

but none of that works (which I'm not surprised because its repackaged).  However, after extensive looking, I still was not able to fund the unms-cli command.  I assumed it would be in there, just at a different path, but I think I've been pretty through, and even tried using the find command, and I am not finding the unms-cli command anywhere.  

I looked for other methods as well.  I think I came close with:
/home/app/unms/cli/user.js

It will list the users in the system, and from viewing the code through "more", it appears to have code to change the password of a user, but I'm not able to decipher how to do so.  I tried all the options used in various forms from the UBNT help page, and got nowhere.  I tried --help, and got the same output if I didn't put any options.  It appears if I supply two options, then the first is processed as a command, but everything I've tried comes back with "unknown command <whatever option 1 was>"

So, does anyone have any info/advice on how to proceed?
Thanks!
BTW: I really love this docker image!  I'm running it under portainer, and so far it has worked great!

**Comment by Nico640** [comment]
Hi,

You came very close ;) this image does not have an unms-cli command, because it's actually only a bash script that executes a command in the appropriate container via docker compose. But because everything is contained in a single container here, the unms-cli script wouldn't work.
However, you can actually look inside the unms-cli script and check what it's actually doing when resetting the password (not exactly straightforward though because you need to get unms-cli from somewhere).

The command you're looking for is:  `node cli/user.js password USERNAME`. Replace USERNAME with the username of your UISP user and execute it inside the container, it should prompt for a new password and change it.

btw, the web gui password recovery only works if you configured an e-mail server in UISP.

**Reply by karnophage** [reply]
Thank you!

**Comment by kusznir** [comment]
Great, thanks!
--Jim
